src/main/io/osd.c

```diff
@@ -437,7 +437,12 @@ static bool osdDrawSingleElement(uint8_t item)
     case OSD_HOME_DIR:
         if (STATE(GPS_FIX) && STATE(GPS_FIX_HOME)) {
             if (GPS_distanceToHome > 0) {
-                const int h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(attitude.values.yaw);
+                int h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(attitude.values.yaw);
+                //If there is no mag sensor, use the heading from GPS chip
+                if(!sensors(SENSOR_MAG)){
+                  h = GPS_directionToHome - DECIDEGREES_TO_DEGREES(gpsSol.groundCourse);
```

Comment:
Instead of adding this in a makeshift way into multiple places in OSD, and leaving all other places to use `attitude.values.yaw`, this should be added to the attitude estimator.

Also, this breaks everybody who does not fly straight line, resulting in bogus information displayed to them, so this should default to off, in order for pilots to acknowledge that this will happen by enabling it.